forge-ai/src/main/java/forge/ai/simulation/SpellAbilityPicker.java

```diff
@@ -353,7 +352,9 @@ private AiPlayDecision canPlayAndPayForSim(final SpellAbility sa) {
         if (!ComputerUtilCost.canPayCost(sa, player, sa.isTrigger())) {
             return AiPlayDecision.CantAfford;
         }
-
+        if (sa.usesTargeting() && !sa.hasLegalTargets()) {
```

Comment:
I have some doubts you can bail out this early in all cases...
I'm assuming targets aren't set up already at this stage, so a call to `StaticAbilityMustTarget.meetsMustTargetRestriction` would make it fail (if such a card was in play).
Secondly is this used in recursion again (as part of a plan chain) with the right game state? I'm thinking about a play where the target for the second spell would only become legal because of the first spell.

Reply:
> I have some doubts you can bail out this early in all cases... I'm assuming targets aren't set up already at this stage, so a call to `StaticAbilityMustTarget.meetsMustTargetRestriction` would make it fail (if such a card was in play). 

Sorry, can you provide an example? I agree that targets wouldn't be set up yet - I guess your point is that `StaticAbilityMustTarget.meetsMustTargetRestriction`  would be a no-op here given it only does something if targets are set up?

> Secondly is this used in recursion again (as part of a plan chain) with the right game state? I'm thinking about a play where the target for the second spell would only become legal because of the first spell.
This should be handled correctly because the game object that's used is already after simulation of any previous effects in the plan.

Reply:
Nah my point is it won't be a no-op as currently the engine doesn't track this process with that in mind

In other words you're assuming it's not caller's responsibility because the timing is determined by the rules but most likely your logic will erroneously make it bail out before any upcoming targeting attempts

(Granted in an ideal world you could safely call this during any game stage but not all parts of Forge are quite that mature :P )

Reply:
Thanks, switched to `ComputerUtilAbility.isFullyTargetable()` so we don't call `StaticAbilityMustTarget.meetsMustTargetRestriction()` here anymore as it's not needed. Remove the helper I have added earlier.